### PR TITLE
Unit conversion for simulationParam

### PR DESCRIPTION
**Pull request by benflexcompute**
- [ ] Added angle unit without auto inferring angle unit for number input.
- [ ]

### CODE REVIEW

File: flow360/component/simulation/unit_system.py [hunk 1]
```diff
@@ -217,12 +217,17 @@ def _unit_inference_validator(value, dim_name, is_array=False):
...
+
     if unit_system_manager.current:
         unit = unit_system_manager.current[dim_name]
         if is_array:
             if all(isinstance(item, Number) for item in value):
                 return value * unit
         if isinstance(value, Number):
+            if dim_name == "angle":
+                raise ValueError(
+                    "Angles must be provided in radians or degrees. Not as pure number."
```
Comment (yifan-flex) [inline flow360/component/simulation/unit_system.py:229]: I think just the first sentence is good enough. Or "explicitly in radians or degrees"

Comment (benflexcompute) [reply flow360/component/simulation/unit_system.py:229]: Removed. Andrzej introduced a new method.

File: flow360/component/simulation/operating_condition.py [hunk 2]
```diff
@@ -142,10 +143,10 @@ class AerospaceCondition(Flow360BaseModel):
 
     # pylint: disable=fixme
     # TODO: add units for angles
-    alpha: float = 0
-    beta: float = 0
+    alpha: AngleType = 0
+    beta: AngleType = 0
```
Comment (yifan-flex) [inline flow360/component/simulation/operating_condition.py:147]: 0 * degree

File: flow360/component/simulation/operating_condition.py [hunk 3]
```diff
@@ -154,8 +155,8 @@ class AerospaceCondition(Flow360BaseModel):
     def from_mach(
         cls,
         mach: pd.PositiveFloat,
-        alpha: float = 0,
-        beta: float = 0,
+        alpha: AngleType = 0,
+        beta: AngleType = 0,
```
Comment (yifan-flex) [inline flow360/component/simulation/operating_condition.py:159]: Same here. 0 * u.degree

**Pull request merged**